go-controller/pkg/ovn/controller/admin_network_policy/admin_network_policy.go

```diff
@@ -32,6 +48,818 @@ func (c *Controller) processNextANPWorkItem(wg *sync.WaitGroup) bool {
+		if err != nil {
+			return fmt.Errorf("failed to create ACL ops: %v", err)
+		}
+		pg := libovsdbops.BuildPortGroup(portGroupName, ports, desiredACLs, pgExternalIDs)
+		ops, err = libovsdbops.CreateOrUpdatePortGroupsOps(c.nbClient, ops, pg)
+		if err != nil {
+			return fmt.Errorf("failed to create ops to add port to a port group: %v", err)
+		}
+		_, err = libovsdbops.TransactAndCheck(c.nbClient, ops)
+		if err != nil {
+			return fmt.Errorf("failed to run ovsdb txn to add ports to port group: %v", err)
+		}
+		// Let us update the anpPriorityMap cache
+		// if this is a fresh ANP add, we will store it in the priority map
+		c.anpPriorityMap[desiredANPState.priority] = anp.Name
+		// since transact was successful we can finally populate the cache
+		c.anpCache[anp.Name] = desiredANPState
+		return nil
+	}
+
+	// ANP state existed in the cache, which means its either an ANP update or pod/namespace add/update/delete
+	klog.V(3).Infof("Admin network policy %s/%d was found in cache...Syncing it", currentANPState.name, currentANPState.priority)
+	// Did the ANP.Spec.Priorty Change? && || Did ANP.Spec.Ingress Change (rule inserts/deletes)? && || Did ANP.Spec.Egress Change (rule inserts/deletes)? && ||
+	// If yes we need to fully recompute the acls present in our ANP's port group; Let's do a full recompute and return.
+	// Reason behind a full recompute: Each rule has precendence based on its position and priority of ANP; if any of that changes
+	// better to delete and recreate ACLs rather than figure out from caches
+	// TODO(tssurya): Investigate if we can be smarter about which ACLs to delete and which to add
+	// rather than always cleaning up everything and recreating them. But this is tricky since rules have precendence
+	// from their ordering. Same goes for ANP Priority update - all rules have to be recreated with the right priorities.
+	// NOTE: Changes to admin policies should be a rare action (can be improved post user feedback) - usually churn would be around namespaces and pods
+	fullPeerRecompute := false
+	if currentANPState.priority != desiredANPState.priority ||
+		len(currentANPState.egressRules) != len(desiredANPState.egressRules) ||
+		len(currentANPState.ingressRules) != len(desiredANPState.ingressRules) {
+		// full recompute
+		// which means delete all ACLs and recreate them
+		klog.V(3).Infof("ANP %s with priority %d was updated", desiredANPState.name, desiredANPState.priority)
+		currentACLs, err := c.buildANPACLs(currentANPState, portGroupName)
+		if err != nil {
+			return fmt.Errorf("unable to build old ACL ops, %v", err)
+		}
+		currentACLsWithUUID, err := libovsdbops.FindACLs(c.nbClient, currentACLs)
+		if err != nil {
+			return fmt.Errorf("failed to find current acls that should be deleted: %w", err)
+		}
+		ops, err = libovsdbops.DeleteACLsFromPortGroupOps(c.nbClient, ops, portGroupName, currentACLsWithUUID...)
+		if err != nil {
+			return fmt.Errorf("unable to create old ACL deletion ops, %v", err)
+		}
```

Comment:
ah I knew there had to be an easier way to do this than what I was doing...but there was a caveat here for that... because BuildPortGroup also wanted LSPs and I didn't want to touch ports at this stage, however like you said this should just be a simple updateOp on the portgroup for the ACLs alone, and having a second update op for ports in the same port-group in the upcoming section should be fine..